tests/rule_based_profiler/config/test_base.py

```diff
@@ -0,0 +1,497 @@
+    data = {
+        "name": "my_RBP",
+        "config_version": 1.0,
+        "rules": {
+            "rule_1": {
+                "name": "rule_1",
+                "domain_builder": {"class_name": "DomainBuilder"},
+                "parameter_builders": [
+                    {"class_name": "ParameterBuilder", "parameter_name": "my_parameter"}
+                ],
+                "expectation_configuration_builders": [
+                    {
+                        "class_name": "ExpectationConfigurationBuilder",
+                        "expectation_type": "expect_column_pair_values_A_to_be_greater_than_B",
+                    }
+                ],
+            },
+        },
+        "author": "Charles Dickens",
+    }
+    schema = RuleBasedProfilerConfigSchema()
+
+    with caplog.at_level(logging.INFO):
+        config = schema.load(data)
+
+    assert isinstance(config, RuleBasedProfilerConfig)
+    assert data["author"] == config.author
+    assert len(caplog.messages) == 1  # author kwarg
+
+
+def test_rule_based_profiler_config_unsuccessfully_loads_with_missing_required_fields():
+    data = {}
+    schema = RuleBasedProfilerConfigSchema()
+
+    with pytest.raises(ValidationError) as e:
+        schema.load(data)
+
+    assert all(
+        f"'{attr}': ['Missing data for required field.']" in str(e.value)
+        for attr in (
+            "name",
+            "config_version",
+            "rules",
+        )
+    )
+
+
+def test_rule_based_profiler_config_successfully_dumps_to_dictionary(caplog):
+    schema = RuleBasedProfilerConfigSchema()
+    config = RuleBasedProfilerConfig(
+        name="my_RBP",
+        config_version=1.0,
+        rules={
+            "rule_1": RuleConfig(
+                name="rule_1",
+                domain_builder=DomainBuilderConfig(class_name="DomainBuilder"),
+                parameter_builders=[
+                    ParameterBuilderConfig(
+                        class_name="ParameterBuilder", parameter_name="my_parameter"
+                    )
+                ],
+                expectation_configuration_builders=[
+                    ExpectationConfigurationBuilderConfig(
+                        class_name="ExpectationConfigurationBuilder",
+                        expectation_type="expect_column_pair_values_A_to_be_greater_than_B",
+                    )
+                ],
+            )
+        },
+        timestamp="2017-02-15|20:26:08.937881",
+    )
+
+    with caplog.at_level(logging.INFO):
+        data = schema.dump(config)
+
+    assert isinstance(data, dict)
+    assert len(data["rules"]) == 1 and isinstance(data["rules"]["rule_1"], dict)
+
+    # As we invoke the serialization/deserialization methods of child objects, their logging messages are also emitted
+    assert all(
+        f"Removed '{attr}' due to null value" in caplog.messages
+        for attr in ("module_name", "mostly", "batch_request")
+    )
```

Comment:
@cdkini Not sure if I saw a test of the `ProfilerConfig` in the way that the fact that it extends `BaseYamlConfig` causes its `@post_load` method must not exist, because it does not get deserialized into an object used by another object (because it is a top-level object, whose configuration can be serialized into a `YAML` file).  Thanks!

Reply:
aha I did not consider the difference due to the `BaseYamlConfig` relationship. Let me look into this some more; I may request your help once I've done some work here.

Reply:
@cdkini Please see `class DataContextConfig(BaseYamlConfig):` and `class CheckpointConfig(BaseYamlConfig):` for analogous examples.

Reply:
Open to add better tests around this particular schema/config